packages/client/src/runtime/core/engines/common/resolveEnginePath.ts

```diff
@@ -0,0 +1,130 @@
+import Debug from '@prisma/debug'
+import { getEnginesPath } from '@prisma/engines'
+import { getNodeAPIName, getPlatform, Platform } from '@prisma/get-platform'
+import { plusX } from '@prisma/internals'
+import fs from 'fs'
+import path from 'path'
+
+import { PrismaClientInitializationError } from '../../errors/PrismaClientInitializationError'
+import { EngineConfig } from './Engine'
+import { binaryTargetsWasIncorrectlyPinned } from './errors/engine-not-found/binaryTargetsWasIncorrectlyPinned'
+import { bundlerHasTamperedWithEngineCopy } from './errors/engine-not-found/bundlerHasTamperedWithEngineCopy'
+import { EngineNotFoundErrorInput } from './errors/engine-not-found/EngineNotFoundErrorInput'
+import { nativeGeneratedOnDifferentPlatform } from './errors/engine-not-found/nativeGeneratedOnDifferentPlatform'
+import { toolingHasTamperedWithEngineCopy } from './errors/engine-not-found/toolingHasTamperedWithEngineCopy'
+
+const debug = Debug('prisma:client:engines:resolveEnginePath')
+
+// this name will be injected by esbuild when we build/bundle the runtime
+const runtimeBuildName = () => (TARGET_ENGINE_TYPE === 'all' ? 'index' : TARGET_ENGINE_TYPE)
+const runtimeFileRegex = () => new RegExp(`${runtimeBuildName()}\\.m?js$`)
```

Comment:
```suggestion
const runtimeFileRegex = () => new RegExp(`runtime[\\\\/]${runtimeBuildName()}\\.m?js$`)
```
So it won't match bundle names that end with `library`, for example.